File: code/section_iii/3_3_insert_single_record.py
from sqlalchemy import create_engine, text

engine = create_engine('sqlite:///thunderbird_manufacturing.db')
conn = engine.connect()

# INSERT a new record 
stmt = text("INSERT INTO CUSTOMER (CUSTOMER_NAME, ADDRESS, CITY, STATE, ZIP, CATEGORY) VALUES (:customerName,:address,:city,:state,:zip,:category)")

conn.execute(stmt, customerName = "Brittania Solutions, LLC",
                       address="2374 Collie Way",
                       city="Frisco",
                       state="TX",
                       zip= 75035,
                       category="COMMERCIAL"
                       )

# Check records to see if last one inserted
for r in conn.execute(text("SELECT * FROM CUSTOMER")):
    print(r)

conn.close()

File: code/section_iii/3_3_insert_single_record.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

public class JavaLauncher {

    public static void main(String[] args) {

        try {

            Connection conn = DriverManager.getConnection("jdbc:sqlite:/c:/git/oreilly_advanced_sql_for_data/thunderbird_manufacturing.db");
            PreparedStatement insertStmt = conn.prepareStatement("INSERT INTO CUSTOMER (CUSTOMER_NAME, ADDRESS, CITY, STATE, ZIP, CATEGORY) VALUES (?,?,?,?,?,?)");

            insertStmt.setString(1, "Brittania Solutions, LLC");
            insertStmt.setString(2, "2374 Collie Way");
            insertStmt.setString(3, "Frisco");
            insertStmt.setString(4, "TX");
            insertStmt.setInt(5, 75035);
            insertStmt.setString(6, "COMMERCIAL");

            // Execute INSERT
            insertStmt.execute();

            // Select all records to check
            PreparedStatement selectStmt = conn.prepareStatement("SELECT * FROM CUSTOMER");
            ResultSet rs = selectStmt.executeQuery();

            while (rs.next()) {
                System.out.println(rs.getInt("CUSTOMER_ID") + "," + rs.getString("CUSTOMER_NAME") + "," + rs.getString("STATE") + "," + rs.getString("CATEGORY"));
            }

            conn.close();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}